packages/tpetra/core/src/Tpetra_Details_PackTraits.hpp

```diff
@@ -192,12 +203,18 @@ struct PackTraits {
       // not have the right size.  However, we require that all
       // entries of the input array have the correct size.
       const size_t numBytes = numEnt * packValueCount (inBuf(0));
-#ifdef HAVE_TPETRA_DEBUG
-      TEUCHOS_TEST_FOR_EXCEPTION(
-        static_cast<size_t> (outBuf.dimension_0 ()) < numBytes,
-        std::invalid_argument, "PackTraits::packArray: outBuf.dimension_0() = "
-        << outBuf.dimension_0 () << " < numBytes = " << numBytes << ".");
-#endif // HAVE_TPETRA_DEBUG
+
+      if (static_cast<size_t> (outBuf.dimension_0 ()) < numBytes) {
+        std::ostringstream os;
+        os << "PackTraits::packArray: outBuf.dimension_0() = "
+           << outBuf.dimension_0 () << " < numBytes = " << numBytes << ".";
+        if (errStr.get() == NULL) {
+          errStr = std::unique_ptr<std::string>(new std::string(""));
+        }
+        *errStr = *errStr + os.str();
+        stat = 1;
+        return 0;
+      }
 
       // FIXME (mfh 02,05 Feb 2015) This may assume UVM.  On the other
```

Comment:
No need for the FIXME any more.  memcpy works as desired on the GPU as well as on the CPU.